### ISSUE TITLE
Find dialog improvement: allow to search by regexp

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#16426]
### Is your feature request related to a problem? Please describe.
Text search is widely used by blind people, more widely than by sighted ones. That's why NVDA has a specific dialog to perform search in a browseable document.
Sometimes however, the text that needs to be found is not a specific word/string but rather a pattern, e.g. a number.

### Describe the solution you'd like
I suggest to add an option in NVDA's find dialog to allow to search specifying a regex rather than a normal text string.

Considering this is a feature targeting advanced users, we may hide it, e.g.:
* put it under a "more options" button in the find dialog
* or displaying this option only if a "Use advanced find dialog" checkbox is checked in the Advanced settings dialog

Or we may leave this option accessible to all, considering that another regex option is already accessible in NVDA's dictionary dialogs.

### Describe alternatives you've considered
Implement this in an add-on. See the following request:
https://github.com/marlon-sousa/EnhancedFindDialog/issues/24

### Additional context
None

**Comment by mltony** [comment]
+1, I would also like to see this feature.

**Comment by XLTechie** [comment]
+1 as well.

My use case, is when searching a large web page or page with a giant table, for several terms that might mean the same thing. It would be most useful to be able to or those terms, and search all at once.

**Comment by ABuffEr** [comment]
+1, absolutely. Together with search history, that briefly appeared time ago.

**Comment by Adriani90** [comment]
Closing as duplicate of #6916 which contains more discussion and even a referenced closed PR which attempted to bring a similar feature to the log viewer.

So feel free to add the arguments and/or give a +1 there.

**Comment by seanbudd** [comment]
Reopening as this issue follows the issue template and more efficiently summaries #6916.
We generally encourage new issues to replace old issues without the template and hard to follow discussion

**Comment by Adriani90** [comment]
Sean that old one contains some important arguments and people are tagged there. Out of my experience of now 6 years of triaging I strongly advise to continue discussions on the older issues with a valuable conversation history, otherwise arguments will go lost on the way.Von meinem iPhone gesendetAm 23.04.2024 um 07:19 schrieb Sean Budd ***@***.***>:﻿
Reopening as this issue follows the issue template and more efficiently summaries #6916.
We generally encourage new issues to replace old issues without the template and hard to follow discussion

**Comment by seanbudd** [comment]
Please see our policy on triaging legacy issues here: 
https://github.com/nvaccess/nvda/blob/master/projectDocs/issues/triage.md#legacy-issues

I'd encourage a separate issue or discussion to suggest changes to this

**Comment by Adriani90** [comment]
Fully fine for me, I think though for developers and people interested to work on implementations a good conversation history is more important than a clean properly filled template without history.But yeah, if this is the way to go we will now start closing lots of issues with really good discussions out there. It was just an advise, but yeah in the end it is your product.Von meinem iPhone gesendetAm 23.04.2024 um 07:55 schrieb Sean Budd ***@***.***>:﻿
Please see our policy on triaging legacy issues here:
https://github.com/nvaccess/nvda/blob/master/projectDocs/issues/triage.md#legacy-issues
I'd encourage a separate issue or discussion to suggest changes to this

**Comment by gerald-hartig** [comment]
@Adriani90 particularly if an issue is several years old, we can @ those contributors that are still known to be active on the project. And if we have linked to the older issue the conversation is not lost. If you still disagree I'm more than happy for us to take this conversation offline and discuss how we can improve our policies/procedures.

**Comment by XLTechie** [comment]
IMO the most important concern from the old issue, is https://github.com/nvaccess/nvda/issues/6916#issuecomment-282524841, from @jcsteh.
Anyone considering working on this should be aware.

**Comment by Adriani90** [comment]
Yes but that‘s your subjective opinion. Commenting on the old one with constructive arguments is also interesting also for others who contributed there. I would also tag @dkager. And as we can see, in this way things get noisy. Lets continue the constructive discussion in the old request. Von meinem iPhone gesendetAm 23.04.2024 um 09:42 schrieb Luke Davis ***@***.***>:﻿
IMO the most important concern from the old issue, is #6916 (comment), from @jcsteh.
Anyone considering working on this should be aware.

**Comment by seanbudd** [comment]
Hi - @XLTechie  we regularly hide comments on noisy discussions that don't contribute to triaging the issue or implementing a solution.
Community support is welcome - it's just something we do tidy up the many "+1" or "bump" comments.

Reopening the issue  - I assume this was a mistake @Adriani90
